### PR TITLE
Fix terrain shader upgrade for URP. Add upgraders for speedtree.

### PR DESCRIPTION
**Pull request by nerites**
# **Please read;**
## **PR Workflow for the Graphics repository:**
* **All PRs must be opened as draft initially**
* Reviewers can be added while the PR is still in draft
* The PR can be marked as “Ready for Review” once the reviewers have confirmed that **no more changes are needed**
* Tests will start automatically after the PR is marked as “Ready for Review”
* **Do not use [skip ci]** - this can break some of our tooling
* Read the [Graphics repository & Yamato FAQ](http://go/graphics-yamato-faq).

### Checklist for PR maker
- [x] Have you added a backport label (if needed)? For example, the `need-backport-*` label. After you backport the PR, the label changes to `backported-*`.
- [x] Have you updated the changelog? Each package has a `CHANGELOG.md` file.
- [ ] Have you updated or added the documentation for your PR? When you add a new feature, change a property name, or change the behavior of a feature, it's best practice to include related documentation changes in the same PR.
- [ ] Have you added a graphic test for your PR (if needed)? When you add a new feature, or discover a bug that tests don't cover, please add a graphic test.

---
### Purpose of this PR
Why is this PR needed, what hard problem is it solving/fixing?

---
### Testing status

**Manual Tests**: What did you do?
- [x] Opened test project + Run graphic tests locally
- [x] Built a player
- [ ] Checked new UI names with UX convention
- [ ] Tested UI multi-edition + Undo/Redo + Prefab overrides + Alignment in Preset
- [ ] C# and shader warnings (supress shader cache to see them)
- [ ] Checked new resources path for the reloader (in developer mode, you have a button at end of resources that check the paths)
- Other: 

**Automated Tests**: What did you setup? (Add a screenshot or the reference image of the test please)

**Yamato**: (Select your branch):
https://yamato.prd.cds.internal.unity3d.com/jobs/902-Graphics

Any test projects to go with this to help reviewers?

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

File: com.unity.render-pipelines.universal/Editor/UniversalRenderPipelineMaterialUpgrader.cs [hunk 1]
```diff
@@ -416,6 +418,21 @@ public TerrainUpgrader(string oldShaderName)
         }
     }
 
+    public class SpeedTreeUpgrader : MaterialUpgrader
```
Comment (phi-lira) [inline com.unity.render-pipelines.universal/Editor/UniversalRenderPipelineMaterialUpgrader.cs:421]: Make it private or internal (we made the other ones public by mistake)

File: com.unity.render-pipelines.universal/Runtime/ShaderUtils.cs [hunk 2]
```diff
@@ -25,6 +27,8 @@ namespace UnityEngine.Rendering.Universal
             "Universal Render Pipeline/Simple Lit",
             "Universal Render Pipeline/Unlit",
             "Universal Render Pipeline/Terrain/Lit",
+            "Universal Render Pipeline/Nature/SpeedTree7",
+            "Universal Render Pipeline/Nature/SpeedTree7 Billboard",
```
Comment (phi-lira) [inline com.unity.render-pipelines.universal/Runtime/ShaderUtils.cs:31]: what about speedtree8?

Comment (nerites) [reply com.unity.render-pipelines.universal/Runtime/ShaderUtils.cs:31]: Ah yeah, I should probably add that as well. The test scene I was using only had SpeedTree7 shaders but I'll add an upgrader for Nature/SpeedTree8 -> "Universal Render Pipeline/Nature/SpeedTree8"

File: com.unity.render-pipelines.universal/Editor/UniversalRenderPipelineMaterialUpgrader.cs (hunk 1 above)
Comment (nerites) [reply com.unity.render-pipelines.universal/Editor/UniversalRenderPipelineMaterialUpgrader.cs:421]: > 
> 
> LGTM. Minor questions and also lacking changelog entry.

Did I add the changelong in the wrong place? https://github.com/Unity-Technologies/Graphics/pull/1350/commits/50f7554d02a6ed2d65b6d0f72f6bb695ef3963d1

File: com.unity.render-pipelines.universal/Runtime/ShaderUtils.cs (hunk 2 above)
Comment (nerites) [reply com.unity.render-pipelines.universal/Runtime/ShaderUtils.cs:32]: Done in https://github.com/Unity-Technologies/Graphics/pull/1350/commits/cae0ba0b7a3c48d28f80f74322ba22707a9581c2

File: com.unity.render-pipelines.universal/Editor/UniversalRenderPipelineMaterialUpgrader.cs (hunk 1 above)
Comment (nerites) [reply com.unity.render-pipelines.universal/Editor/UniversalRenderPipelineMaterialUpgrader.cs]: Made internal in https://github.com/Unity-Technologies/Graphics/pull/1350/commits/cae0ba0b7a3c48d28f80f74322ba22707a9581c2

**Comment by phi-lira** [comment]
PR LGTM, However ABV is not green.

**Comment by nerites** [comment]
I had to update an editor test, but all checks are green now! :)

**Pull request merged**